tests/bluetooth/ctrl_isoal/src/main.c

```diff
@@ -20,1517 +20,7747 @@
-struct isoal_sink *sink;
-
-/*
- * PDU defines and helper functions
- */
-#define TESTDATA_MAX_LEN 60
-uint8_t testdata[TESTDATA_MAX_LEN] = {
-	0xBF, 0x9B, 0xD1, 0x7D, 0x9E, 0xE0, 0xB9, 0xA4, 0x71, 0xE6,
-	0x80, 0xA7, 0x59, 0xAD, 0xB0, 0x98, 0xB3, 0x95, 0x83, 0x6B,
-	0xF8, 0xFE, 0xCB, 0xA1, 0xE9, 0x7A, 0xDD, 0x86, 0x68, 0x50,
-	0x77, 0x6E, 0xF2, 0x5C, 0xD7, 0x53, 0x62, 0x56, 0x74, 0x89,
-	0xCE, 0xC5, 0xAA, 0x65, 0x8C, 0xFF, 0xF5, 0x8F, 0xDA, 0xBC,
-	0xB6, 0xEC, 0x5F, 0xE3, 0xD4, 0xEF, 0xC8, 0x92, 0xC2, 0xFB
-};
+#define TEST_RX_PDU_PAYLAOD_MAX	(40)
+#define TEST_RX_PDU_SIZE		(TEST_RX_PDU_PAYLAOD_MAX + 2)
 
-uint8_t pdu_data[(TESTDATA_MAX_LEN+2)];
-struct node_rx_iso_meta meta;
-struct isoal_pdu_rx pdu_meta = {&meta, (struct pdu_iso *) &pdu_data};
+#define TEST_RX_SDU_FRAG_PAYLOAD_MAX	(100)
 
 #define LLID_TO_STR(llid) (llid == PDU_BIS_LLID_COMPLETE_END ? "COMPLETE_END" : \
 	(llid == PDU_BIS_LLID_START_CONTINUE ? "START_CONT" : \
 		(llid == PDU_BIS_LLID_FRAMED ? "FRAMED" : \
 			(llid == PDU_BIS_LLID_CTRL ? "CTRL" : "?????"))))
 
+#define DU_ERR_TO_STR(err) (err == 1 ? "Bit Errors" : \
+			(err == 2 ? "Data Lost" : \
+			(err == 0 ? "OK" : "Undefined!")))
+
 #define STATE_TO_STR(s) (s == BT_ISO_SINGLE ? "SINGLE" : \
 	(s == BT_ISO_START ? "START" : \
 		(s == BT_ISO_CONT ? "CONT" : \
 			(s == BT_ISO_END ? "END" : "???"))))
 
-#define DEBUG_TEST 1
+#define ROLE_TO_STR(s) (s == BT_ROLE_BROADCAST ? "Broadcast" :			\
+	(role == BT_CONN_ROLE_PERIPHERAL ? "Peripheral" :			\
+	(role == BT_CONN_ROLE_CENTRAL ? "Central" : "Undefined")))
+
+#define FSM_TO_STR(s) (s == ISOAL_START ? "START" : \
+	(s == ISOAL_CONTINUE ? "CONTINUE" : \
+		(s == ISOAL_ERR_SPOOL ? "ERR SPOOL" : "???")))
+
+
+struct rx_pdu_meta_buffer {
+	struct isoal_pdu_rx pdu_meta;
+	struct node_rx_iso_meta meta;
+	uint8_t pdu[TEST_RX_PDU_SIZE];
+};
+
+struct rx_sdu_frag_buffer {
+	uint16_t write_loc;
+	uint8_t sdu[TEST_RX_SDU_FRAG_PAYLOAD_MAX];
+};
+
+/**
+ * Intializes a RX PDU buffer
+ * @param[in] buf Pointer to buffer structure
+ */
+static void init_rx_pdu_buffer(struct rx_pdu_meta_buffer *buf)
+{
+	memset(buf, 0, sizeof(struct rx_pdu_meta_buffer));
+	buf->pdu_meta.meta = &buf->meta;
+	buf->pdu_meta.pdu = (struct pdu_iso *) &buf->pdu[0];
+}
+
+/**
+ * Initializes a RX SDU buffer
+ * @param[in] buf Pointer to buffer structure
+ */
+static void init_rx_sdu_buffer(struct rx_sdu_frag_buffer *buf)
+{
+	memset(buf, 0, sizeof(struct rx_sdu_frag_buffer));
+}
 
 #if defined(DEBUG_TEST)
-void debug_print_pdu(void)
+/**
+ * Print contents of PDU
+ * @param[in] pdu_meta Pointer to PDU structure including meta information
+ */
+static void debug_print_rx_pdu(struct isoal_pdu_rx *pdu_meta)
 {
-	uint8_t len = pdu_data[1];
+	zassert_not_null(pdu_meta, "");
 
-	PRINT("PDU %04d : %10s %02d: ", (uint32_t) meta.payload_number,
-		LLID_TO_STR(pdu_data[0]), len);
+	PRINT("PDU %04d (%10d) : %12s [%10s] %03d: ",
+		(uint32_t) pdu_meta->meta->payload_number,
+		(uint32_t) pdu_meta->meta->timestamp,
```

Comment:
In Zephyr we generally do not have spaces between the type and the value when casting. 
```suggestion
		(uint32_t)pdu_meta->meta->payload_number,
		(uint32_t)pdu_meta->meta->timestamp,
```